### PR TITLE
fix(qwen3_moe): step-0 NaN in MXFP8 packed finetune — expert unload + fused RoPE

### PR DESCRIPTION
**Pull request by hemildesai**
Two independent bugs make the Qwen3-MoE-30B MXFP8/packed-sequence finetune produce NaN at step 0 (and garbage loss in bf16). Both are fixed here.

## Bug 1 — GroupedExpertsTE experts silently unloaded (`state_dict_mixin.py`)

`GroupedExpertsTE` (`backend.experts == "te"`) exposes `gate_and_up_projs` / `down_projs` as a **`torch.stack` COPY** of TransformerEngine's per-expert `weight{i}` params — it does **not** alias the model's grouped expert storage.

At `ep_shard == 1` (`ep_size == world_size`), the merged-expert state-dict conversion marks the key as in-place-loadable, so DCP does `target.copy_(source)` through strided views. For the TE backend those views point at the throwaway `torch.stack`, so the data never reaches TE's real `weight{i}` params — the experts stay at **random init**, **silently** (`strict=False`). bf16 → garbage loss (~15.8); MXFP8 → NaN.

Fix: exclude `experts == "te"` from the in-place path so it rebuilds via `_set_stacked_weight`. Other backends (`GroupedExperts`, `GroupedExpertsDeepEP`) keep a real aliasing `nn.Parameter` and are unaffected. #2682 closed only the **quantization** sub-case of this same hazard; a plain **bf16** checkpoint still slipped through.

## Bug 2 — fused TE RoPE NaNs the backward on packed (THD) sequences (3 configs)

Fused TransformerEngine RoPE is numerically unstable on THD/packed sequences: the forward stays finite, but the **backward emits NaN gradients** at step 0 (`grad_norm nan`), which NaNs training from step 1; under MXFP8 it surfaces as a step-0 forward NaN. The packed recipes don't set `rope_fusion`, so they inherit the default (`HAVE_TE and cuda` → `True`) and hit the bad path.

Fix: pin `rope_fusion: false` in the `model.backend` block of the three affected packed recipes (`qwen3_moe_30b_te_hybridep_mxfp8`, `qwen3_moe_30b_te_hybridep`, `qwen3_moe_30b_te_packed_sequence`). Mirrors #2687, which applied the identical one-line fix to the sibling `qwen3_moe_30b_te_packed_sequence_lora` recipe; see #2674. (The other packed qwen3-moe recipes — `magi_*`, `flashoptim`, `lora` — already set it.)

## Verification

GB200 Qwen3-MoE-30B (Qwen3-30B-A3B) finetune, `ep_size=8`, packed sequences. With both fixes, bf16, TE-MXFP8, and torchao-MXFP8 all train cleanly and converge identically:

| precision | step-0 loss | step-0 grad_norm | → step-2 loss |
|---|---|---|---|
| bf16 | 3.04 | 13.88 | 2.23 |
| TE-MXFP8 | 3.04 | 13.81 | 2.22 |
| torchao-MXFP8 | 3.05 | 15.69 | 2.23 |

bf16 ran the full 50 steps (loss 3.04 → 1.89, grad_norm → ~1.1, zero NaN). The MXFP8 forward-NaN was downstream of the fused-RoPE corruption — there is no separate MXFP8 kernel/block-scaling issue.

Refs nvbug 6350293.

### CODE REVIEW

File: nemo_automodel/components/moe/state_dict_mixin.py [hunk 1]
```diff
@@ -644,6 +644,13 @@ def _convert_single_merged_expert_to_hf_split_experts(
         # rebuild path: never mark these keys in-place-loaded when ``quantization`` is set.
         quantization = kwargs.get("quantization", False)
 
+        # GroupedExpertsTE (backend.experts == "te") exposes gate_and_up_projs/down_projs as a
+        # torch.stack COPY of TE's per-expert weight{i} params; it does not alias the model's
+        # grouped storage, so an in-place copy_ would write that throwaway buffer and leave the TE
+        # experts at their initial values. Force the rebuild path for it. Other expert backends
+        # keep a real aliasing stacked Parameter and are unaffected.
+        experts_alias_grouped_storage = getattr(getattr(self, "backend", None), "experts", None) != "te"
```
[the remark this answers is not in the document]
Comment (hemildesai) [reply nemo_automodel/components/moe/state_dict_mixin.py:652]: Good catch — added regression tests in `test_state_dict_mixin.py`: `test_inplace_load_skips_when_backend_experts_is_te_gate_and_up` and `..._down_projs` assert that with `backend.experts == "te"` the load emits contiguous copies and does not register the native key as in-place-loaded (i.e. takes the rebuild path), for both projections.

**Review by akoumpa — DISMISSED** [review]

**Review by akoumpa — APPROVED** [review]
